### PR TITLE
gui: help browser

### PR DESCRIPTION
**Pull request by gadfort**
Adds:
- `HelpBrowser` widget to allow for interactive / offline viewing of help if it's available.

Changes:
- Adds two functions to OpenROAD to allow for a centralized location to determine the location of the docs
- `man` switched to use new path access.

Notes:
- ~Due to the exclusion of `std::filesystem` I had to do path and directory handling manually, this can be simplified if `std::filesystem` is allowed.~
- `man` could be configured to open the gui help if we want.
- The HTML help is not particularly pretty and I don't find it helpful to know the compile date or header, but these issues can be handled separately.
- The script widget could be modified to allow for hyperlinks to this widget got commands and error messages (that would be best handled in a separate PR).
- there are lots of missing functions from this (outside the scope of this PR), but anything in ODB, ORD, and STA are completely missing.
- the GUI commands probably just need to be promoted out of the namespace from `gui::show` -> `gui_show` or `show_gui` since the man page generation clobbers its name (should be done in a separate PR).

Preview:
image

### CODE REVIEW

**Comment by maliberty** [comment]
Could you include a screen shot or two?  

Now that we are off centos7 I think we can try to use std filesystem and hopefully not having linking headaches.

File: src/OpenRoad.cc [hunk 1]
```diff
@@ -640,6 +640,44 @@ int OpenRoad::getThreadCount()
...
+  const std::string exe = getExePath();
+
+  if (exe.empty()) {
+    return "";
+  }
+
+  // remove binary name
+  std::string path = exe.substr(0, exe.find_last_of('/'));
+
+  const std::size_t build_idx = exe.length() - 13;
```
Comment (maliberty) [inline src/OpenRoad.cc:666]: why 13?

File: src/gui/src/gui.i [hunk 2]
```diff
@@ -741,4 +741,13 @@ void unminimize()
   gui->unminimize();
 }
 
+void show_help(const std::string& item)
```
Comment (maliberty) [inline src/gui/src/gui.i:744]: Add to README?

**Comment by maliberty** [comment]
Based on #6241 you can use std filesystem

**Comment by gadfort** [comment]
@maliberty I'll rework the file handling then to use std::filesystem

File: src/OpenRoad.cc (hunk 1 above)
Comment (gadfort) [reply src/OpenRoad.cc:666]: It's len("/src/OpenROAD"), but should be moot with std::filesystem

File: src/OpenRoad.cc [hunk 3]
```diff
@@ -640,6 +641,40 @@ int OpenRoad::getThreadCount()
...
+  if (exe.empty()) {
+    return "";
+  }
+
+  std::filesystem::path path(exe);
+
+  // remove binary name
+  path = path.parent_path();
+
+  if (path.stem() == "src") {
```
Comment (maliberty) [inline src/OpenRoad.cc:669]: When does this occur?

Comment (gadfort) [reply src/OpenRoad.cc:669]: this happens when executing from the build directory instead of an installed version of openroad

**Review by maliberty — APPROVED** [review]

**Pull request merged**